src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionXpathSingleFilter.java

```diff
@@ -407,6 +407,45 @@
  * &lt;/module&gt;
  * </pre>
  * <p>
+ * Example for trailing comments check to suppress specific trailing comment:
+ * </p>
+ * <pre>
+ * public class Test {
+ *   int a; // SUPPRESS CHECKSTYLE, OK
+ *   int b; // NOPMD, OK
+ *   int c; // NOSONAR, OK
+ *   int d; // violation, not suppressed
+ * }
+ * </pre>
+ * <p>
+ * To configure check so that trailing comment with exact comments like "SUPPRESS CHECKSTYLE, OK",
+ * "NOPMD, OK", "NOSONAR, OK" are suppressed:
+ * </p>
+ * <pre>
+ * &lt;module name="TrailingComment"/&gt;
+ * &lt;module name="SuppressionXpathSingleFilter"&gt;
+ *   &lt;property name="checks" value="TrailingCommentCheck"/&gt;
+ *   &lt;property name="query" value="//SINGLE_LINE_COMMENT
+ *       [./COMMENT_CONTENT[@text=' NOSONAR, OK\n' or @text=' NOPMD, OK\n'
+ *       or @text=' SUPPRESS CHECKSTYLE, OK\n']]"/&gt;
```

Comment:
So we are required to have leading spaces, the same as in comment?
@romani ping

Reply:
Yes, we got no other choice since the comment contains this space. @romani Please share your views

Reply:
@AkMo3 , please remove ` OK` from examples. to keep them exactly as it is used in real life.
yes, leading space is required as user has to provide whole content of comment. If he use both variants of comments, user will have to extend/update Xpath expression to his needs.

Reply:
Done. Teamcity to be fixed in #10305